pallets/settlement/src/lib.rs

```diff
@@ -466,6 +466,8 @@ decl_error! {
         SignerAlreadyExists,
         /// Signer is not added to venue.
         SignerDoesNotExist,
+        /// Instruction leg amount can't be zero
+        InsufficientAmount,
```

Comment:
Could we change to `ZeroAmount` as I think that will be a bit clearer as to what the issue is.